### PR TITLE
fix: python client

### PR DESCRIPTION
**Pull request by pratikmishra356**
## Problem
Describe the problem you are trying to solve here

## Solution
Provide a brief summary of your solution so that reviewers can understand your code

## Environment variable changes

What ENVs need to be added or changed

## Pre-deployment activity
Things needed to be done before deploying this change (if any)

## Post-deployment activity
Things needed to be done after deploying this change (if any)

## API changes
| Endpoint        | Method           | Request body  | Response Body |
| ------------- |:-------------:| -----:| ----------------:|
| API      | GET/POST, etc | request | response |

## Possible Issues in the future
Describe any possible issues that could occur because of this change

### CODE REVIEW

File: clients/python/main.py [hunk 1]
```diff
@@ -1,22 +1,19 @@
 from expclient import ExperimentationClient
-from cacclient import CacClient 
+from cacclient import CacClient , MergeStrategy
 import http.server
 import socketserver
 
 try:
-    tenant_name = "dev"
+    tenant_name = "test"
```
Comment (mahatoankitkumar) [inline clients/python/main.py:7]: Let it be `dev` for consistency around all clients

File: clients/python/main.py [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
     cac_client = CacClient(tenant_name, polling_frequency, cac_host_name)
 
-    exp_client.create_new_experimentation_client()
-    cac_client.create_new_cac_client()
-
     cac_client.start_cac_polling_update()
     exp_client.start_experimentation_polling_update()
 
-    PORT = 8002
+    PORT = 8003
```
Comment (mahatoankitkumar) [inline clients/python/main.py:16]: Okay with this, but any reason to change this?

File: crates/cac_client/src/lib.rs [hunk 3]
```diff
@@ -99,6 +99,9 @@ impl Client {
         let reqwc = clone_reqw(&reqw)?;
         let resp = reqwc.send().await.map_err_to_string()?;
         let last_modified_at = get_last_modified(&resp);
+        if resp.status().is_client_error() {
+            return Err("Invalid tenant given".to_string());
```
Comment (mahatoankitkumar) [inline crates/cac_client/src/lib.rs:103]:
```suggestion
            return Err("Invalid tenant".to_string());
```

**Review by mahatoankitkumar — DISMISSED** [review]

File: clients/python/main.py (hunk 2 above)
Comment (pratikmishra356) [reply clients/python/main.py:16]: No, I will change

File: clients/python/expclient/client.py [hunk 4]
```diff
@@ -15,6 +16,33 @@
...
+
+# x = AttrDict({'some': 'value'})
+# x['blah'] = 'hello world'
+# print(x.blah)  # hello world
+# print(x.some)  # value
+
+# # you can also assign value this way
+# x.foo = 'bar'
+# print(x['foo'])   # bar
+
```
Comment (Datron) [inline clients/python/expclient/client.py:45]: remove comments

File: crates/experimentation_client/src/interface.rs [hunk 5]
```diff
@@ -211,17 +211,20 @@ pub extern "C" fn expt_get_satisfied_experiments(
...
+                    .await
+                    .map(|exp| {
+                        rstring_to_cstring(serde_json::to_value(exp).unwrap().to_string())
+                            .into_raw()
+                    }),
+                std::ptr::null_mut()
+            )
+        }
+    })
 }
```
Comment (Datron) [inline crates/experimentation_client/src/interface.rs:228]: @pratikmishra356 we can't use EXP_RUNTIME for this, it needs localSet for this. Can we keep the same code, just unwrap safely and return that

File: crates/experimentation_client/src/interface.rs [hunk 6]
```diff
@@ -250,30 +253,33 @@ pub extern "C" fn expt_get_filtered_satisfied_experiments(
...
```
Comment (Datron) [inline crates/experimentation_client/src/interface.rs:270]: Same as above

**Review by Datron — APPROVED** [review]

**Review by ShubhranshuSanjeev — APPROVED** [review]

**Pull request merged**